qualtran/bloqs/arithmetic/sorting_test.py

```diff
@@ -27,19 +30,54 @@ def test_bitonic_sort(bloq_autotester):
 
 
 def test_comparator_manual():
-    bloq = Comparator(4)
-    assert bloq.t_complexity().t == 88 - 4
-    with pytest.raises(NotImplementedError):
-        bloq.decompose_bloq()
+    bloq = Comparator(2**4)
+    assert bloq.t_complexity().t == 88 - 4 - 7 * 4
+
+
+def test_comparator_symbolic_t_complexity():
+    bloq = _cmp_symb.make()
+    bitsize = ceil(log2(bloq.L - 1))
+    assert bloq.t_complexity() == TComplexity(t=15 * bitsize + 4, clifford=56 * bitsize + 24)
+
+
+@pytest.mark.parametrize("L", [5, 8, 12])
+def test_comparator_classical_sim(L: int):
+    bloq = Comparator(L)
+    for a in range(L):
+        for b in range(L):
+            res_a, res_b, anc = bloq.call_classically(a=a, b=b)
+            assert res_a <= res_b
 
 
 def test_bitonic_sort_manual():
     bitsize = 4
     k = 8
-    bloq = BitonicSort(bitsize, k)
-    assert bloq.t_complexity().t == 8 * 9 * (88 - 4)
-    with pytest.raises(NotImplementedError):
-        bloq.decompose_bloq()
+
+    bloq = BitonicSort(2**bitsize, k)
+    assert bloq.num_comparisons == 24
+
+    _ = bloq.t_complexity()
+
+
+def test_bitonic_sort_classical_sim():
+    L = 8
+    xs = np.array([4, 2, 7, 1])
+    bloq = BitonicSort(L, len(xs))
+    sorted_xs, _ = bloq.call_classically(xs=xs)
+    assert np.all(sorted_xs == sorted(xs))
+
+
```

Comment:
your new bloqs should add autotesters and bloq-doc-specs too.

Reply:
added bloq_examples with autotesters and updated the bloqdocspecs